### PR TITLE
boards: mimxrt1050_evk: Elaborate doc to be more user-facing

### PR DESCRIPTION
**Pull request by pfalcon**
1. Change "just the image" of the board with the image which shows
component location (taken from user guide PDF). This is especially
useful, as the board doesn't have descriptive silkscreen labels,
only component numberings, e.g. there're 3 microUSB connectors
labeled just Jxx, Jyy, Jzz.
2. Mention more MCU features useful to a user.
3. Elaborate the "Memory" section, to describe memory structure to
provide complete enough picture to a user.
4. Explicitly mention that the board (currently) doesn't have any
means to program flash, and the only way to run apps is by running
them via debugger.
5. Explicitly mention that a fresh board received requires installing
different USB bootloader firmware.
6. Standardize on referring to "ninja" as a build tool, as the rest
of the docs does.
7. Other minor changes and clarifications.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

### CODE REVIEW

**Comment by pfalcon** [comment]
@dbkinder: A few suggestions to you based on experience with this docs:

1. To get everyone on the same line, and provide information useful, not confusing, to a user, can you please pay attention that all board docs specify memory sizes in kilobytes/megabytes/etc., not kilobits/megabits?
2. Can you please pay attention that the board image size? Here, I pureposed scaled/compressed the image to be not bigger than the previous image, and the latter one was quite good being under 100K. I guess that's the guideline we should have, while I saw 0.5MB board images. With hundred(s) of boards shipped, that becomes a concern, we don't want to ship users 50+MB of useless images (a user usually interested in one, maybe a few boards). Nor we want to leave such file sizes in git history either. Pinging @nashif regarding this particular point too.

Both points are probably worth being added to board docs guidelines, if we have them, or if not, it's worth to start them probably ;-).

**Comment by dbkinder** [comment]
I took a look at the data sheet (https://www.mouser.com/datasheet/2/302/iMX-RT1050-EVKFS-1381468.pdf) and it would appear that the original author of the doc wrote it, she transcribed the datasheet's "256 Mb" (megabits) as "256 MB" (megabytes) - simple mistake.

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by pfalcon** [comment]
> transcribed the datasheet's "256 Mb" (megabits) as "256 MB" (megabytes)

There was "KB" in this doc actually...

> (The data sheet incorrectly mixes use of Kb and KB itself!)

Yes, happens all the time in the real world. We could try to be better here in Zephyr.

**Comment by dbkinder** [comment]
Some recommended doc edits still need to be resolved.

**Comment by pfalcon** [comment]
@dbkinder :

> Some recommended doc edits still need to be resolved.

Ack, still in my "do it when code freeze is announced" backlog, working on code things in the meantime.

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 1]
```diff
@@ -182,14 +187,23 @@ remaining are not used.
 Programming and Debugging
 *************************
 
+**NOTE:** MCU on this board lacks builtin Flash memory. Programming external
```
[the remark this answers is not in the document]
Comment (MaureenHelm) [reply boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: This note isn't accurate anymore, we can now program external flash with jlink tools. I need to rework this section for all the rt boards.

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 2]
```diff
@@ -182,14 +187,23 @@ remaining are not used.
...
+Flash memory using Zephyr tools is not yet supported. The way to get an
+application running is starting it in debug mode using ``ninja debug`` (see
+below).
+
 The MIMXRT1050-EVK includes the :ref:`nxp_opensda` serial and debug adapter
 built into the board to provide debugging, flash programming, and serial
 communication over USB.
 
+The board comes pre-programmed with mbedRTOS compatible bootloader firmware,
```
[the remark this answers is not in the document]
Comment (MaureenHelm) [reply boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: I'd rather this say: `The board comes pre-programmed with DAPLink firmware.`

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 3]
```diff
@@ -26,13 +26,17 @@ MIMXRT1050-EVKB board, refer to `Board Revisions`_ section.
...
 - Memory
 
-  - 256 KB SDRAM
-  - 64 Mbit QSPI Flash
-  - 512 Mbit Hyper Flash
+  - 512 KByte MCU-internal SRAM
+  - 32 MByte external SDRAM (166MHz)
+  - 8 MByte external QSPI Flash
+  - 64 MByte external Hyper Flash
+  - 96 KByte MCU BootROM for loading application from external memory
```
Comment (MaureenHelm) [inline boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: I prefer "KB" and "MB", but will defer to @dbkinder

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Comment by MaureenHelm** [comment]
Also, please don't change the image.

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst (hunk 3 above)
Comment (dbkinder) [reply boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: my preference too (KB, MB)

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbkinder) [reply boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: @MaureenHelm I think the ball is in your court for updating this...

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by MaureenHelm** [comment]
> @MaureenHelm I think the ball is in your court for updating this...

@dbkinder yes, it is. see #13584

**Comment by dleach02** [comment]
Can this PR be closed?

**Comment by nashif** [comment]
stale...

**Pull request closed without merging**